extendr-macros/src/wrappers.rs

```diff
@@ -381,7 +390,15 @@ fn translate_meta_arg(input: &mut FnArg, self_ty: Option<&syn::Type>) -> syn::Re
         FnArg::Typed(ref mut pattype) => {
             let pat = pattype.pat.as_ref();
             let ty = pattype.ty.as_ref();
-            let name_string = quote! { #pat }.to_string();
+            let pat_ident = if let syn::Pat::Ident(ref pat_ident) = pat {
+                &pat_ident.ident
+            } else {
+                return Err(syn::Error::new_spanned(
+                    input,
+                    "failed to translate name of argument",
+                ));
```

Comment:
Do you mind swapping this to a `match` statement like you have about? Its easier to understand 😅 

Are these `syn` patterns that you are now catching that previously weren't?

Reply:
Done